src/main/java/org/springframework/integration/kafka/inbound/KafkaMessageSource.java

```diff
@@ -0,0 +1,660 @@
+		}
+
+		@Override
+		public boolean isAutoAck() {
+			return this.autoAckEnabled;
+		}
+
+	}
+
+	/**
+	 * Information for building an KafkaAckCallback.
+	 *
+	 * @param <K> the key type.
+	 * @param <V> the value type.
+	 *
+	 */
+	public static class KafkaAckInfo<K, V> implements Comparable<KafkaAckInfo<K, V>>, ProducerFactory<K, V> {
+
+		private final Object consumerMonitor;
+
+		private final String groupId;
+
+		private final Consumer<K, V> consumer;
+
+		private final ConsumerRecord<K, V> record;
+
+		private final TopicPartition topicPartition;
+
+		private final AtomicReference<Producer<K, V>> producer;
+
+		private final Map<TopicPartition, Set<KafkaAckInfo<K, V>>> offsets;
+
+		private final AtomicBoolean myTransaction;
+
+		private volatile boolean rolledBack;
+
+		private volatile boolean ackDeferred;
+
+		KafkaAckInfo(Object consumerMonitor, String groupId, Consumer<K, V> consumer, ConsumerRecord<K, V> record,
+				TopicPartition topicPartition, AtomicReference<Producer<K, V>> producerRef,
+				Map<TopicPartition, Set<KafkaAckInfo<K, V>>> offsets, AtomicBoolean myTransaction) {
+			this.consumerMonitor = consumerMonitor;
+			this.groupId = groupId;
+			this.consumer = consumer;
+			this.record = record;
+			this.topicPartition = topicPartition;
+			this.producer = producerRef;
+			this.offsets = offsets;
+			this.myTransaction = myTransaction;
+		}
+
+		Object getConsumerMonitor() {
+			return this.consumerMonitor;
+		}
+
+		public String getGroupId() {
+			return this.groupId;
+		}
+
+		public Consumer<K, V> getConsumer() {
+			return this.consumer;
+		}
+
+		public ConsumerRecord<K, V> getRecord() {
+			return this.record;
+		}
+
+		public TopicPartition getTopicPartition() {
+			return this.topicPartition;
+		}
+
+		public Producer<K, V> getProducer() {
+			return this.producer.get();
+		}
+
+		public Map<TopicPartition, Set<KafkaAckInfo<K, V>>> getOffsets() {
+			return this.offsets;
+		}
+
+		public boolean isRolledBack() {
+			return this.rolledBack;
+		}
+
+		public void setRolledBack(boolean rolledBack) {
+			this.rolledBack = rolledBack;
+		}
+
+		public boolean isAckDeferred() {
+			return this.ackDeferred;
+		}
+
+		public void setAckDeferred(boolean ackDeferred) {
+			this.ackDeferred = ackDeferred;
+		}
+
+		public boolean isMyTransaction() {
+			return this.myTransaction.get();
+		}
+
+		@Override
+		public int compareTo(KafkaAckInfo<K, V> other) {
+			return Long.compare(this.record.offset(), other.getRecord().offset());
+		}
+
+		@Override
+		public String toString() {
+			return "KafkaAckInfo [record=" + this.record + ", rolledBack=" + this.rolledBack + ", ackDeferred="
+					+ this.ackDeferred + ", myTransaction=" + this.myTransaction + "]";
+		}
+
+		@Override
+		public Producer<K, V> createProducer() {
+			Assert.state(this.producer.get() != null, "There is no transactional producer here");
+			return this.producer.get();
+		}
+
+		@Override
+		public boolean transactionCapable() {
+			return true;
```

Comment:
Why is this unconditional?
Looks like I really might not be interested in transactions at all...
Thanks